web/modules/custom/farm_nfa/farm_nfa.module

```diff
@@ -275,6 +275,13 @@ function farm_nfa_form_alter(&$form, FormStateInterface $form_state) {
   }
 }
 
+/**
```

Comment:
Instead of attaching the library to every page, attach it to the map. There are a few ways to do that:

- You can attach it to the map render array like you did with behavior_farm_nfa_gfw_layers in ForestPlanGfwForm.php
- You can create a map behavior config item, and add it to the map in onMapRender. See for example farm_map.map_behavior.farm_nfa_map_activities.yml
- You can create a map behavior config item and include it in the map type config item. See for example farm_map.map_type.farm_nfa_plan_locations.yml

Since dealing with configuration in forests is complicated I would be incline to go with option 1, attaching the library to the render array. But please do it whichever way you like best.

Reply:
@peacog initially i also went with option 1, but the issues that was happening with it was, the library was not getting properly attached on individual location assets pages such as `http://forests.ddev.site/asset/472` etc. hence i went with the hook approach, but i guess option 2 seems like a nicer approach so will update this accordingly, thanks !